java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -78,6 +80,29 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
     return result;
   }
 
+  public SymbolicValue createMethodSymbolicValue(MethodInvocationTree syntaxNode, List<SymbolicValue> values) {
+    SymbolicValue result;
+    if (isEqualsMethod(syntaxNode)) {
+      result = new SymbolicValue.MethodEqualsToSymbolicValue(counter);
+      result.computedFrom(values);
+    } else {
+      result = createDefaultSymbolicValue(syntaxNode);
+    }
+    counter++;
+    return result;
+  }
+
+  private static boolean isEqualsMethod(MethodInvocationTree syntaxNode) {
+    if (syntaxNode.arguments().size() == 1) {
```

Comment:
Checking the argument number is not enough to tell that you are calling the equals method inherited and potentially overridden from `Object`. What about an object having a method such as:
```
public boolean equals(int i) { 
   ...
}
```
You should also check that the called method uses `Object` as first parameter.